tools/reproducible.Dockerfile

```diff
@@ -97,10 +97,12 @@ FROM builder AS build
 
 ARG LEAP_BUILD_JOBS
 
-COPY / /src
-RUN cmake -S src -B build -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -GNinja && \
+# Yuck: This places the source at the same location as leap's CI (build.yaml, build_base.yaml). Unfortunately this location only matches
+#       when build.yaml etc are being run from a repository named leap.
+COPY / /__w/leap/leap
+RUN cmake -S /__w/leap/leap -B build -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -GNinja && \
     cmake --build build -t package -- ${LEAP_BUILD_JOBS:+-j$LEAP_BUILD_JOBS} && \
-    src/tools/tweak-deb.sh build/leap_*.deb
+    /__w/leap/leap/tools/tweak-deb.sh build/leap_*.deb
```

Comment:
And in the perfect example for failure of,
> It's important these two methods remain functionally identical for the produced .deb file to ensure that both CI & manual user builds produce the exact same output

This change was required for the above statement to remain true because in some cases full paths get baked in to the final executable. When built via `build-base.yaml` the source path is effectively `/__w/leap/leap/` while the reproducible.Dockerfile was `/src/`. I entertained 3 ways to fix this,

1. Change CI to place source in `/src`. This involves changes in many locations because we `actions/checkout@v3` in many different jobs during the build & tests.
2. Change CI to make use of something like `-ffile-prefix-map=$PWD=/src`. This seems to work, but results in more variation of compile flags between CI & manual build.
3. Change reproducible.Dockerfile's build stage to place source in `/__w/leap/leap` like CI does.

Given the 11th hour, I opted for option 3 even though it has a negative trade off that CI only places source in `/__w/leap/leap` if the repository is named `leap`. If someone were to fork the repo to a different name, they would have divergence between CI & manual reproducible builds. But our current CI workflows are, unfortunately, ENF centric (see #416 for how this could be resolved; though I doubt something like the NP Tests will pass on a free GH runner these days), so this seems like an okay tradeoff for the time being. We can migrate to option 2 in the future is really needed.